[Ling Guan Ming] iP

CeereeC opened the pull request:
## DukePro
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
>
DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)
________________________________________________________________________________________
If you Java programmer, you can use it to practice Java too. Here's the main method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

CeereeC committed:
Level 1. Greet, Echo, Exit

CeereeC committed:
Level 2. Add, List

CeereeC committed:
Level 3. Mark as Done

CeereeC committed:
Level-4: ToDo, Event, Deadline

CeereeC committed:
A-TextUiTesting: Automated Text UI Testing

CeereeC committed:
Level 5. Handle Errors

CeereeC committed:
Level-6: Delete

CeereeC committed:
Implement Level 7

CeereeC committed:
Implement Level-8

CeereeC committed:
Fix merge conflicts between Level-7 and Level-8

CeereeC committed:
Implement A-MoreOOP

CeereeC committed:
Implement A-Packages

CeereeC committed:
Merge branch 'add-gradle-support'

CeereeC committed:
Implement A-Gradle

CeereeC committed:
Implement A-JUnit

CeereeC committed:
Implement A-Jar

CeereeC committed:
Implement A-JavaDoc

CeereeC committed:
Implement A-CodingStandard

CeereeC committed:
Implement Level 9

waynezsy commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,29 @@
+import duke.ui.Ui;
+
+/**
+ * FIndCommand class
```
I think you accidentally capitalised the I.

waynezsy commented on src/main/java/duke/data/Storage.java:
```diff
@@ -0,0 +1,119 @@
+    public List<Task> load() {
+        List<Task> newList = new ArrayList<>();
+        try {
+            Scanner s = new Scanner(file);
```
Perhaps you may want to make the Scanner name slightly longer than just "s" as it can be slightly confusing since the method also deals heavily with Strings? I think simply making it "sc" would also help improve clarity.

waynezsy commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,13 @@
+package duke;
```
Is there a reason this class is being tracked on revision control? This feels like a class created to test your gradle test functionalities rather than code meant for your product so I think this isn't meant to be included?

waynezsy reviewed (commented):
Your code feels very clean :)
Most of your variable names are also extremely descriptive, good job!
I don't think I can really find any problems based on the course's naming convention guidelines :)

zanwenfu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+package duke;
+
+import duke.command.Command;
+import duke.common.exception.DukeException;
+import duke.data.Storage;
+import duke.task.TaskList;
+import duke.ui.Parser;
+import duke.ui.Ui;
```
Good that you separate package import declaration, also import classes in detail.

zanwenfu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
```
Good that you make class variables private.

zanwenfu commented on src/main/java/duke/command/IncorrectCommand.java:
```diff
@@ -0,0 +1,22 @@
+ */
+public class IncorrectCommand extends Command {
+
+    public final String feedbackToUser;
```
Class variable here should be private?

zanwenfu commented on src/main/java/duke/data/Storage.java:
```diff
@@ -0,0 +1,119 @@
+                    task = new Event(desc, strArr[3]);
+                    break;
+                case "D":
+                    task = new Deadline(desc, LocalDate.parse(strArr[3]));
+                    break;
+                default:
+                    throw new IOException();
+                }
+                if (strArr[1].equals("true")) {
+                    TaskList.markAsDone(task);
+                }
+                newList.add(task);
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println(e.getMessage());
+        } catch (IOException e) {
+            System.out.println("Error with input file");
+        } catch (DateTimeParseException e) {
+            System.out.println("Deadline format is incorrect");
+        }
+        return newList;
+    }
+
+    /**
+     * Stores the TaskList in the local disk storage
+     *
+     * @param taskList ArrayList of tasks to be saved
+     */
+    public void store(TaskList taskList) {
+        String dataList = taskList.getDataList();
+        try {
+            PrintWriter pw = new PrintWriter(file);
+            pw.print(dataList);
+            pw.close();
+        } catch (FileNotFoundException e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+}
```
Overall looks good!

zanwenfu reviewed (commented):
Clean code! Good job.

CeereeC committed:
Implement A-CheckStyle

CeereeC committed:
Implement Level-10

CeereeC committed:
Implement A-Assertions

CeereeC committed:
Implement A-CodeQuality

CeereeC committed:
Merge pull request #1 from CeereeC/branch-A-Assertions

Implement A-Assertions

CeereeC committed:
Merge pull request #2 from CeereeC/branch-A-CodeQuality

Implement A-CodeQuality

CeereeC committed:
Implement BCD-Extension

Implemented B-ViewSchedules
 as a way to view tasks grouped by their schedules.
This is so we can have a new filtered view.

CeereeC committed:
Add ui image

CeereeC committed:
Add Jar file

CeereeC committed:
Implement A-UserGuide

CeereeC committed:
Fix UI
